# Introdução à Ciência de Dados - UFPB
Professor: Yuri Malheiros

## Medidas de Centralidade e Dispersão

A partir do dataset `movies.csv` calcule a média, mediana e desvio padrão para as colunas:

In [15]:
import pandas as pd
pd.options.display.float_format = "{:.3f}".format

movies_df = pd.read_csv("../../caiov/ICD/movies.csv")
movies_df

Title     US Gross  Worldwide Gross  US DVD Sales  \
0                 The Land Girls   146083.000       146083.000           nan   
1         First Love, Last Rites    10876.000        10876.000           nan   
2     I Married a Strange Person   203134.000       203134.000           nan   
3           Let's Talk About Sex   373615.000       373615.000           nan   
4                           Slam  1009819.000      1087521.000           nan   
...                          ...          ...              ...           ...   
3196  Zack and Miri Make a Porno 31452765.000     36851125.000  21240321.000   
3197                      Zodiac 33080084.000     83080084.000  20983030.000   
3198                        Zoom 11989328.000     12506188.000   6679409.000   
3199         The Legend of Zorro 45575336.000    141475336.000           nan   
3200           The Mask of Zorro 93828745.000    233700000.000           nan   

      Production Budget Release Date MPAA Rating  Running Time (min)  \
0           8000000.000    12-Jun-98           R                 nan   
1            300000.000     7-Aug-98           R                 nan   
2            250000.000    28-Aug-98         NaN                 nan   
3            300000.000    11-Sep-98         NaN                 nan   
4           1000000.000     9-Oct-98           R                 nan   
...                 ...          ...         ...                 ...   
3196       24000000.000    31-Oct-08           R             101.000   
3197       85000000.000     2-Mar-07           R             157.000   
3198       35000000.000    11-Aug-06          PG                 nan   
3199       80000000.000    28-Oct-05          PG             129.000   
3200       65000000.000    17-Jul-98       PG-13             136.000   

             Distributor                        Source        Major Genre  \
0               Gramercy                           NaN                NaN   
1                 Strand                           NaN              Drama   
2              Lionsgate                           NaN             Comedy   
3              Fine Line                           NaN             Comedy   
4                Trimark           Original Screenplay              Drama   
...                  ...                           ...                ...   
3196       Weinstein Co.           Original Screenplay             Comedy   
3197  Paramount Pictures     Based on Book/Short Story  Thriller/Suspense   
3198       Sony Pictures  Based on Comic/Graphic Novel          Adventure   
3199       Sony Pictures                        Remake          Adventure   
3200       Sony Pictures                        Remake          Adventure   

             Creative Type         Director  Rotten Tomatoes Rating  \
0                      NaN              NaN                     nan   
1                      NaN              NaN                     nan   
2                      NaN              NaN                     nan   
3                      NaN              NaN                  13.000   
4     Contemporary Fiction              NaN                  62.000   
...                    ...              ...                     ...   
3196  Contemporary Fiction      Kevin Smith                  65.000   
3197         Dramatization    David Fincher                  89.000   
3198            Super Hero     Peter Hewitt                   3.000   
3199    Historical Fiction  Martin Campbell                  26.000   
3200    Historical Fiction  Martin Campbell                  82.000   

      IMDB Rating  IMDB Votes  
0           6.100    1071.000  
1           6.900     207.000  
2           6.800     865.000  
3             nan         nan  
4           3.400     165.000  
...           ...         ...  
3196        7.000   55687.000  
3197          nan         nan  
3198        3.400    7424.000  
3199        5.700   21161.000  
3200        6.700    4789.000  

[3201 rows x 16 columns]

### US Gross

In [16]:
# ADICIONE O SEU CÓDIGO AQUI

Gross = {"Média": movies_df["US Gross"].mean(),
                  "Mediana": movies_df["US Gross"].median(),
                  "Desvio Padrão": movies_df["US Gross"].std()}
gross = pd.Series(Gross)

Gross_Values = pd.DataFrame({"US Gross": gross})
Gross_Values

US Gross
Média         44002085.164
Mediana       22019465.500
Desvio Padrão 62555311.391

### Production Budget

In [17]:
# ADICIONE O SEU CÓDIGO AQUI

PB = {"Média": movies_df["Production Budget"].mean(),
                  "Mediana": movies_df["Production Budget"].median(),
                  "Desvio Padrão": movies_df["Production Budget"].std()}
pb = pd.Series(PB)

PB_Values = pd.DataFrame({"Production Budget": pb})
PB_Values

Production Budget
Média               31069171.448
Mediana             20000000.000
Desvio Padrão       35585913.445

### Running Time

In [18]:
# ADICIONE O SEU CÓDIGO AQUI

RT = {"Média": movies_df["Running Time (min)"].mean(),
                  "Mediana": movies_df["Running Time (min)"].median(),
                  "Desvio Padrão": movies_df["Running Time (min)"].std()}
rt = pd.Series(RT)

RT_Values = pd.DataFrame({"Running Time": rt})
RT_Values

Running Time
Média               110.194
Mediana             107.000
Desvio Padrão        20.171

### IMDB Rating

In [19]:
# ADICIONE O SEU CÓDIGO AQUI

IMDB_R = {"Média": movies_df["IMDB Rating"].mean(),
                  "Mediana": movies_df["IMDB Rating"].median(),
                  "Desvio Padrão": movies_df["IMDB Rating"].std()}
imdb_r = pd.Series(IMDB_R)

IMDB_R_Values = pd.DataFrame({"IMDB Rating": imdb_r})
IMDB_R_Values

IMDB Rating
Média                6.283
Mediana              6.400
Desvio Padrão        1.252

### IMDB Votes

In [20]:
# ADICIONE O SEU CÓDIGO AQUI

IMDB_V = {"Média": movies_df["IMDB Votes"].mean(),
                  "Mediana": movies_df["IMDB Votes"].median(),
                  "Desvio Padrão": movies_df["IMDB Votes"].std()}
imdb_v = pd.Series(IMDB_V)

IMDB_V_Values = pd.DataFrame({"IMDB Votes": imdb_v})
IMDB_V_Values

IMDB Votes
Média           29908.645
Mediana         15106.000
Desvio Padrão   44937.582

### Análise dos Resultados

Quais colunas tiveram valores muito diferentes entre média e mediana? Quais os seus desvios padrão?

**As colunas US Gross (62555311.391 - DV), Production Budget (35585913.445 - DV) e a IMDB Votes (44937.582 - DV).** 

Que conclusões podemos tirar desses resultados?

**Que os valores dessa coluna em relação a sua média são um pouco distantes um do outro, resultando em um alto valor de desvio padrão. Mostra também que provavelmente os valores mínimo e máximo de cada coluna também são distantes um do outro. Além disso caso não tivéssemos o desvio padrão neste Data Frame provavelmente o entendimento apenas com a média seria prejudicado, já que esta não revela a realidade dos valores da coluna.**